x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -576,6 +579,10 @@ protected Clock getClock() {
         );
     }
 
+    public Map<String, Processor.Factory> getProcessors(Processor.Parameters parameters) {
+        return Collections.singletonMap(InferenceProcessor.TYPE, new InferenceProcessor.Factory());
```

Comment:
A secondary idea is that we don't have a single inference processor. Instead, we have an individual processor for each supported model type. 

This MAY make sense, depending on how the models will be gathered and cached.